### PR TITLE
AlenaBurak/7-answers-for-users

### PR DESCRIPTION
**Pull request by ElenaBurak**
Test without correct answers for users

### CODE REVIEW

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs [hunk 1]
```diff
@@ -47,11 +47,23 @@ public async Task<TestDTO> GenerateTestAsync(GenerateTestDTO generateTestDTO)
             var grammarQuestionsDTO = _mapper.Map<GrammarQuestionDTO[]>(grammarQuestions);
             var auditionQuestionsDTO = _mapper.Map<AuditionQuestionDTO[]>(auditionQuestions);
             var topicsDTO = _mapper.Map<TopicQuestionDTO[]>(topics);
+            foreach (var item in grammarQuestionsDTO)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Services/QuestionService.cs]: linq

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs [hunk 2]
```diff
@@ -47,11 +47,23 @@ public async Task<TestDTO> GenerateTestAsync(GenerateTestDTO generateTestDTO)
...
+            {
+                foreach (var z in item.Answers)
+                {
+                    z.IsCorrect = null;
+                }
+            }
 
-            
+            foreach (var item in auditionQuestionsDTO)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Services/QuestionService.cs]: same

File: exalearn-api/ExaLearn.Bl/Services/QuestionService.cs (hunk 1 above)
Comment (romanstat) [reply exalearn-api/ExaLearn.Bl/Services/QuestionService.cs]: try to not use foreach

**Review by vlad12345123 — APPROVED** [review]

File: exalearn-api/ExaLearn.Bl/Services/TestService.cs [hunk 3]
```diff
@@ -65,9 +65,16 @@ public async Task<AssessmentDTO> CreateAssesmentAsync(AssessmentDTO assessmentDT
             assessment.General = (assessment.Audition + assessment.Grammar + assessment.Speaking + assessment.Essay) / assessmentCount;
             await _assessmentRepository.SaveChangesAsync();
             passedTest.AssessmentId = assessment.Id;
-            await _passedTestRepository.UpdateAsync(passedTest);
-
             passedTest.Status = StatusType.Checked;
+            await _passedTestRepository.UpdateAsync(passedTest);
+           
+            if (assessment.General >= 5)
```
Comment (romanstat) [inline exalearn-api/ExaLearn.Bl/Services/TestService.cs]: >5? So little

**Review by romanstat — APPROVED** [review]

**Review by vlad12345123 — APPROVED** [review]

**Pull request merged**